akka-stream/src/main/scala/akka/stream/impl/JsonObjectParser.scala

```diff
@@ -107,21 +107,22 @@ import akka.util.ByteString
     val bufSize = buffer.length
 
     skipToNextObject(bufSize)
+    val maxObjectLengthIndex = if (maximumObjectLength == Int.MaxValue) Int.MaxValue else pos + maximumObjectLength
```

Comment:
Good point. I wonder if the code can made to work in those cases, where leading bytes to be skipped plus object will not fit into an array?

We should definitely fix the condition here but might also want to add a guard to `offer` to avoid creating a new array with negative size and also throw the FramingException from below.